libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -247,6 +247,13 @@ wl_status_t ESP8266WiFiSTAClass::begin() {
  * @param dns1       Static DNS server 1
  * @param dns2       Static DNS server 2
  */
+
+#if LWIP_VERSION_MAJOR != 1
+#undef netif_set_addr // need to call lwip-v1.4 netif_set_addr()

```

Comment:
This is confusing to me. The define above means that lwip2 is in play, then this gets undef'd with a comment about using the lwip1.4 function, then a function is imported from extern.
Is an lwip1.4 function really being used? If so, is there now a dependency on both lwips?
Why the undef then import? is there a C-style name clash, so this is done to change the function?